### PR TITLE
Classes made from Struct should have default `new` singleton method.

### PR DESCRIPTION
**Pull request by marcandre**
~~Please see ruby-core thread~~

Many existing PRs & ruby-core threads, everyone agrees on the correct behavior too 😄

### CODE REVIEW

**Comment by ko1** [comment]
> ruby-core thread
which thread?

**Comment by marcandre** [comment]
Ah ah, looks like I'm late to the party 😆 

I looked at the other patches, they indeed all do the same. I amended mine a bit according to what I saw and added everyone as co-authors, hopefully this will win the most co-authored commit ever!

**Comment by marcandre** [comment]
> > ruby-core thread
> 
> which thread?

Didn't have time to create it! You guys are way too quick for me 🤣

**Comment by marcandre** [comment]
So unless there's objection or other modifications I'll commit this and we'll only need a backport

File: struct.c [hunk 1]
```diff
@@ -316,25 +316,16 @@ rb_struct_s_inspect(VALUE klass)
...
 }
 
-static VALUE
-struct_new_kw(int argc, const VALUE *argv, VALUE klass)
-{
-    return rb_class_new_instance_kw(argc, argv, klass, RB_PASS_CALLED_KEYWORDS);
-}
-
 static VALUE
 setup_struct(VALUE nstr, VALUE members, int keyword_init)
```
Comment (nobu) [inline struct.c:320]: Maybe `keyword_init` argument is no longer used?

Comment (marcandre) [reply struct.c:320]: `keyword_init` still has an effect in `Struct#initialize` (aka `rb_struct_initialize_m`), but shouldn't have an impact on `new` (which should simply allocate the object an call `initialize`).

Or did I misunderstand your comment?

Comment (marcandre) [reply struct.c:320]: Oh, I see. You are right, we no longer need the argument.

Comment (marcandre) [reply struct.c]: Commit amended 👍

**Comment by nobu** [comment]
Appveyor fails probably due to exceeding `clone_depth: 10`.
Could you rebase once?

**Comment by marcandre** [comment]
@nobu Rebased

**Comment by marcandre** [comment]
Merged! Thanks for the super quick reviews @jeremyevans and @nobu 👍

**Pull request merged**